py-polars/polars/internals/frame.py

```diff
@@ -2493,6 +2493,14 @@ def hstack(
         else:
             return wrap_df(self._df.hstack([s.inner() for s in columns]))
 
+    @tp.overload
+    def vstack(self, df: "DataFrame", in_place: Literal[True]) -> None:
+        ...
+
+    @tp.overload
+    def vstack(self, df: "DataFrame", in_place: Literal[False] = ...) -> "DataFrame":
+        ...
```

Comment:
```suggestion
    @tp.overload
    def vstack(self, df: "DataFrame", in_place: Literal[False] = ...) -> "DataFrame":
        ...


    @tp.overload
    def vstack(self, df: "DataFrame", in_place: bool) -> Optional["DataFrame"]:
        ...
```

Reply:
If I add this in, then the snipped posted above gives:
```console
$ mypy t.py 
t.py:7: note: Revealed type is "None"
t.py:8: note: Revealed type is "polars.internals.frame.DataFrame"
t.py:9: note: Revealed type is "Union[polars.internals.frame.DataFrame, None]"
```

Reply:
I think this is a reasonable compromise for bools / in-place, it does add some clutter to the code base having to write three overloads per method though.

Note that similar issues with strings have been raised before: https://github.com/pola-rs/polars/pull/1937#discussion_r759971391, where unfortunately the fall back means any string is accepted.

Reply:
> I think this is a reasonable compromise for bools / in-place, it does add some clutter to the code base having to write three overloads per method though.

Yes, hopefully we don't have too many return type changing arguments.